gguf-py/scripts/gguf-convert-endian.py

```diff
@@ -82,10 +85,11 @@ def convert_byteorder(reader: gguf.GGUFReader, args: argparse.Namespace) -> None
             delta = tensor.data[block_offs:block_offs + 2].view(dtype=np.uint16)
             delta.byteswap(inplace=True)
             if block_num % 100000 == 0:
-                print(f"[{(n_blocks - block_num) // 1000}K]", end="")
-                sys.stdout.flush()
-        print()
-    print("* Completion")
+                log_message += f"[{(n_blocks - block_num) // 1000}K]"
+
+        logger.info(log_message)
+
+    logger.info("* Completion")
```

Comment:
This is clearly intended to be live progress output. If you are averse to using print for this (even with file=sys.stderr), use tqdm (which calls sys.stderr.write for you).

Reply:
resolved, with specific focus on refactoring this section with emphases on why there is a special handling of the Q8_0 bock and to more easily add nested progress bars